flocx_market/tests/unit/common/test_prepareservice.py

```diff
@@ -0,0 +1,58 @@
+import flocx_market.conf as conf
+import socket
+import flocx_market.common.service as flocx_market_service
+import pytest
+
+CONF = conf.CONF
+
+
+def test_service_defaults():
+    flocx_market_service.prepare_service()
+    assert(CONF.api.host_ip == "0.0.0.0")
+    assert(CONF.api.port == 8081)
+    assert(CONF.api.max_limit == 1000)
+    assert(CONF.api.public_endpoint is None)
+    assert(CONF.api.api_workers is None)
+    assert(CONF.api.enable_ssl_api is False)
+    assert(CONF.flask.SQLALCHEMY_TRACK_MODIFICATIONS is False)
+    assert(CONF.flask.PROPAGATE_EXCEPTIONS is False)
+    assert(CONF.host == socket.gethostname())
+
+    CONF.clear()
```

Comment:
Calling `CONF.clear()` at the end of the test isn't going to work reliably, because in the event of a failure in any of the above `assert` statements, your `CONF.clear()` will not get called (which will in turn cause subsequent tests to fail).  Can you call `CONF.clear()` at the *beginning* of each test?

Reply:
I don't believe calling clear at the beginning will work as any call to prepare_service will modify the state of CONF in some way.

Reply:
I think it *has* to be called at the beginning of the function, because as I pointed out, calling it at the end is problematic. It's okay that prepare_service modifies it; the goal is simply to clear out the changes made by a *prior* call to prepare_service.

Reply:
(You could handle this by using a pytest fixture that would call `CONF.clear` before each test; that would make the implementation a little cleaner but the effect would be the same).